client.go

```diff
@@ -179,7 +177,6 @@ func (c *Client) status(status int) {
 
 // Connect attempts to connect the client to the server.
 func (c *Client) Connect() error {
-	c.status(DIALING)
```

Comment:
@JulianKropp Please add back the DIALING state.

Reply:
If there was a reason for removing the DIALING state please let me know. But I find it useful to differentiate when the client is dialing and when it is actually TCP connected and waiting for the DDP connection protocol to complete.

Reply:
I think the word `DIALING` is a little bit confusing. I would replace it with `CONNECTING`

Reply:
@JulianKropp thanks for the changes. I think DIALING should be familiar to Go developers (it's the standard name for connecting sockets). Most importantly though, we have to keep DIALING as the connecting status enum for backward compatibility for existing library usage. This change otherwise doesn't break backward compatibility and can be a micro release which I greatly prefer. Adding new enums with new states shouldn't break existing code. So if you don't mind changing it to DIALING as the enum name that would great. Thanks!

Reply:
Ok thanks for the explanation. I'm rather new to go. I changed it back